### DISCUSSION TITLE
Is node supposed to support machine going to sleep?

### DISCUSSION THREAD

**Discussion by nazar-pc** [paritytech/substrate#9535] [Q&A] [answered]
I have a somewhat vague question regarding the expectations of the environment the substrate node is running in.

We observe strange issues when the machine node is running on goes to sleep (primarily Docker on Windows on a laptop), after which it doesn't re-establish network connections and has other side effects wich epoch transitions because of that.

Is it really a supported use case or is it expected that the node is running on a machine that doesn't go to sleep?

I can imagine all kinds of assumptions about time to be implicitly baked in in various places that would break when suddenly operations that should take milliseconds take days in practice.

**Comment by bkchr** [comment, marked as answer]
What other side effects? Basically it should disconnect from all connected nodes and start to reconnect.

**Reply by nazar-pc** [reply]
After some fixes in the implementation I think it was all caused by the same bug, so nothing specific to mention here.

**Reply by wilwade** [reply]
@nazar-pc What were the fixes you ended up needing? Seeing the same issue perhaps and curious what fixed it for you.

**Reply by nazar-pc** [reply]
We do not have epochs anymore in our protocol, so BABE-specific issue went away. I'm not aware of other issues of this nature in our protocol at the moment.

**Comment by apopiak** [comment]
I remember seeing issues with block production if BABE is used with a single-node (dev) network IIRC.

**Reply by bkchr** [reply]
Yes with BABE you need to produce one block per session.
